### PR TITLE
`GenesisBuild<T,I>` deprecated. `BuildGenesisConfig` added

### PR DESCRIPTION
**Pull request by michalkucharczyk**
Companion for: paritytech/substrate#14306

Step towards: https://github.com/paritytech/substrate/issues/13334

### CODE REVIEW

**Review by serban300 — APPROVED** [review]
The bridges part looks good.

**Comment by gilescope** [comment]
For test chains it might be ok to use `default()` but we generally prefer everything to be very explicit for value bearing runtimes.

File: polkadot-parachain/src/chain_spec/asset_hubs.rs [hunk 1]
```diff
@@ -227,7 +228,10 @@ fn asset_hub_polkadot_genesis(
 				.map(|k| (k, ASSET_HUB_POLKADOT_ED * 4096))
 				.collect(),
 		},
-		parachain_info: asset_hub_polkadot_runtime::ParachainInfoConfig { parachain_id: id },
+		parachain_info: asset_hub_polkadot_runtime::ParachainInfoConfig {
+			parachain_id: id,
+			..Default::default()
```
[the remark this answers is not in the document]
Comment (bkchr) [reply polkadot-parachain/src/chain_spec/asset_hubs.rs:233]: These are not being used in production. For production we have the json files.

**Review by bkchr — APPROVED** [review]

**Review by skunert — APPROVED** [review]

**Comment by michalkucharczyk** [comment]
bot fmt

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3033832

**Review by davxy — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3133450

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3133489

**Review by andresilva — APPROVED** [review]

**Comment by andresilva** [comment]
@michalkucharczyk once if you fix the `Cargo.lock` conflict it should be ready to merge.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3170521

**Comment by michalkucharczyk** [comment]
bot merge

**Pull request merged**